subprojects/docs/src/docs/userguide/tasks.xml

```diff
@@ -290,52 +290,673 @@
-                <output args="transform" ignoreExtraLines="true"/>
-            </sample>
-            <para>Notice that Gradle executes this task a second time, and does not skip the task even though nothing has
-                changed. Our example task was defined using an action closure. Gradle has no idea what the closure does and
-                cannot automatically figure out whether the task is up-to-date or not. To use Gradle's up-to-date checking,
-                you need to declare the inputs and outputs of the task.
-            </para>
-            <para>Each task has an <literal>inputs</literal> and <literal>outputs</literal> property, which you use to
-                declare the inputs and outputs of the task. Below, we have changed our example to declare that it takes
-                the source XML file as an input and produces output to a destination directory. Let's run it a couple
-                of times.
+
+            <figure>
+                <title>Example task inputs and outputs</title>
+                <imageobject>
+                    <imagedata fileref="img/taskInputsOutputs.png" width="160mm"/>
+                </imageobject>
+            </figure>
+
+            <para>An important characteristic of an input is that it affects one or more outputs, as
+                you can see from the previous figure. Different bytecode is generated depending on the
+                content of the source files and the minimum version of the Java runtime you want to 
+                run the code on. That makes them task inputs. But whether the compilation runs in a
+                forked process or not, determined by the <literal>fork</literal> property, has no
+                impact on what bytecode gets generated. So <literal>fork</literal> is a plain task
+                property.
             </para>
-            <sample id="incrementalTask" dir="userguide/tasks/incrementalBuild/inputsAndOutputs" title="Declaring the inputs and outputs of a task">
-                <sourcefile file="build.gradle"/>
-                <output args="transform" outputFile="generatorTask.out" ignoreExtraLines="true"/>
-                <output args="transform" ignoreExtraLines="true"/>
-            </sample>
-            <para>Now, Gradle knows which files to check to determine whether the task is up-to-date or not.</para>
-            <para>
-                The task's <literal>inputs</literal> property is of type <apilink class="org.gradle.api.tasks.TaskInputs"/>.
-                The task's <literal>outputs</literal> property is of type <apilink class="org.gradle.api.tasks.TaskOutputs"/>.
+            <para>As part of incremental build, Gradle tests whether any of the task inputs or outputs
+                have changed since the last build. If they haven’t, Gradle can consider the task up to
+                date and therefore skip it. Also note that incremental build won’t work unless a task
```

Comment:
Why? Is there a difference between skipping a task and skipping the execution of its actions?